### PR TITLE
NIFI-1865 Close StreamThrottler when processor stops.

### PR DESCRIPTION
**Pull request by ijokarumawak**
Add close when processor stops.

### CODE REVIEW

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/java/org/apache/nifi/processors/standard/TestPostHTTP.java [hunk 1]
```diff
@@ -407,4 +407,31 @@ public void testSendChunked() throws Exception {
...
+        final String suppliedMimeType = "text/plain";
+        runner.setProperty(PostHTTP.URL, server.getUrl());
+        runner.setProperty(PostHTTP.CONTENT_TYPE, suppliedMimeType);
+        runner.setProperty(PostHTTP.CHUNKED_ENCODING, "false");
+        runner.setProperty(PostHTTP.MAX_DATA_RATE, "10kb");
+
+        final Map<String, String> attrs = new HashMap<>();
+        attrs.put(CoreAttributes.MIME_TYPE.key(), "text/plain");
+
+        runner.enqueue(StringUtils.repeat("This is the song that never ends. It goes on and on my friend.", 100).getBytes(), attrs);
```
Comment (mattyb149) [inline nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/java/org/apache/nifi/processors/standard/TestPostHTTP.java]: These song lyrics are copyrighted :-/ Need to replace with some other generic sample text

**Comment by ijokarumawak** [comment]
Thanks for reviewing @mattyb149 , I will fix the texts and resend a PR shortly.

**Comment by alopresto** [comment]
@ijokarumawak for future reference, you can push the changes to the same branch and the PR will update automatically. If you decide to rebase, you can do that as well and follow it with `git push --force`.

**Comment by ijokarumawak** [comment]
@alopresto I heard that single commit would be easier to apply to master and 0.x branch. I squashed the commits and sent a PR #432 . I'll update the same branch and PR when I do rebase, thank you.

**Comment by mattyb149** [comment]
You're both right :) you can do the squash then a rebase and force push to the same PR. Doesn't matter to me though, I'll grab the latest and review/merge, thanks!

**Comment by ijokarumawak** [comment]
I didn't know that I can push --force a squashed commit, thanks!

**Comment by alopresto** [comment]
Yes, some people prefer to `squash` before opening the PR; others open the PR with incremental commits in place to show history/decisions. There are different opinions as to which is easier to review. I believe our [Contributor Guide](https://cwiki.apache.org/confluence/display/NIFI/Contributor+Guide#ContributorGuide-RebasingandSquashingwhileunderreview) allows for both options. If the PR is not squashed, the committer can always do that (and should) before merging (after review).

**Comment by mattyb149** [comment]
May the --force be with you ;)

**Pull request closed without merging**